server/controllers/admin/users/index.js

```diff
@@ -81,12 +82,34 @@ function list(req, res, next) {
   const sql = 'SELECT user.id, display_name, user.username, user.deactivated FROM user;';
 
   db.exec(sql)
-    .then((rows) => {
-      res.status(200).json(rows);
+    .then((users) => {
+      return setRoles(users);
+    }).then(users => {
+
+      res.status(200).json(users);
     })
     .catch(next);
 }
 
+async function setRoles(users) {
+  const sql = `
+    SELECT r.label
+    FROM user_role ur
+    JOIN role r ON r.uuid = ur.role_uuid
+    WHERE ur.user_id = ?`;
+
+  return q.all(users.map(r => db.exec(sql, r.id))).then(roles => {
+    users.forEach((user, index) => {
```

Comment:
Does this really make an SQL query for every user in the database?

A more efficient way would be to do a LEFT JOIN on roles and then use [`GROUP_CONCAT()`](https://dev.mysql.com/doc/refman/5.5/en/group-by-functions.html#function_group-concat) (which https://github.com/IMA-WorldHealth/bhima-2.X/pull/3339 just taught me about).  Could you try that way?

It should be just a couple extra lines of code instead of a whole new function.

Reply:
this function ```js setRoles``` is called in the call list where every users are selected

Reply:
the setRoles is useful because it convert the roles array for a user into string